---
layout: post
title: Transcriptome Assembly - Geoduck Tissue-specific Assembly (Juvenile)
date: '2019-02-15 22:45'
tags:
  - juvenile
  - geoduck
  - Panopea generosa
  - mox
  - transcriptome
  - mox
categories:
  - Miscellaneous
---
Based on a [discussion in this GitHub Issue](https://github.com/RobertsLab/resources/issues/576), I've initiated some tissue-specific transcriptome assemblies with our current geoduck data.

Job was run on Mox and ```rsync```ed to my folder on Gannet.

FastA index files were generated separately via ```samtools faidx Trinity.fasta``` (didn't think about it at the time so did not add to SBATCH script).


SBATCH script:

- [20190215_geo_trinity_juvD5.sh](https://raw.githubusercontent.com/RobertsLab/sams-notebook/master/sbatch_scripts/20190215_geo_trinity_juvD5.sh)(GitHub)

<pre><code>
#!/bin/bash
## Job Name
#SBATCH --job-name=trinity_20190215
## Allocation Definition
#SBATCH --account=coenv
#SBATCH --partition=coenv
## Resources
## Nodes
#SBATCH --nodes=2
## Walltime (days-hours:minutes:seconds format)
#SBATCH --time=5-00:00:00
## Memory per node
#SBATCH --mem=120G
##turn on e-mail notification
#SBATCH --mail-type=ALL
#SBATCH --mail-user=samwhite@uw.edu
## Specify the working directory for this job
#SBATCH --workdir=/gscratch/scrubbed/samwhite/outputs/20190215_trinity_geoduck_juvD5_RNAseq

# Load Python Mox module for Python module availability

module load intel-python3_2017

# Document programs in PATH (primarily for program version ID)

date >> system_path.log
echo "" >> system_path.log
echo "System PATH for $SLURM_JOB_ID" >> system_path.log
echo "" >> system_path.log
printf "%0.s-" {1..10} >> system_path.log
echo ${PATH} | tr : \\n >> system_path.log

data_dir=/gscratch/scrubbed/samwhite/data/P_generosa/RNAseq
trinity_dir=/gscratch/srlab/programs/Trinity-v2.8.3
assembly_stats=assembly_stats.txt

# Run Trinity
${trinity_dir}/Trinity \
--trimmomatic \
--seqType fq \
--max_memory 120G \
--CPU 56 \
--left \
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-1_S8_L001_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-2_S16_L002_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-3_S24_L003_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-4_S32_L004_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-5_S40_L005_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-6_S48_L006_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-7_S56_L007_R1_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-8_S64_L008_R1_001.fastq.gz \
--right \
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-1_S8_L001_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-2_S16_L002_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-3_S24_L003_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-4_S32_L004_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-5_S40_L005_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-6_S48_L006_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-7_S56_L007_R2_001.fastq.gz,\
${data_dir}/Geoduck-larvae-day5-RNA-EPI-99-8_S64_L008_R2_001.fastq.gz

# Assembly stats
${trinity_dir}/util/TrinityStats.pl trinity_out_dir/Trinity.fasta \
> ${assembly_stats}
</code></pre>

---

#### RESULTS

Output folder:

- [20190215_trinity_geoduck_juvD5_RNAseq/](http://gannet.fish.washington.edu/Atumefaciens/20190215_trinity_geoduck_juvD5_RNAseq/)

Trinity assembly:

- [20190215_trinity_geoduck_juvD5_RNAseq/trinity_out_dir/Trinity.fasta](http://gannet.fish.washington.edu/Atumefaciens/20190215_trinity_geoduck_juvD5_RNAseq/trinity_out_dir/Trinity.fasta)(FastA)

  - FastA index (```samtools faidx```):

    - [20190215_trinity_geoduck_juvD5_RNAseq/trinity_out_dir/Trinity.fasta.fai](http://gannet.fish.washington.edu/Atumefaciens/20190215_trinity_geoduck_juvD5_RNAseq/trinity_out_dir/Trinity.fasta.fai)

Assembly stats:

- [20190215_trinity_geoduck_juvD5_RNAseq/assembly_stats.txt](http://gannet.fish.washington.edu/Atumefaciens/20190215_trinity_geoduck_juvD5_RNAseq/assembly_stats.txt)

---

<pre><code>


################################
## Counts of transcripts, etc.
################################
Total trinity 'genes':	235162
Total trinity transcripts:	402320
Percent GC: 36.64

########################################
Stats based on ALL transcript contigs:
########################################

	Contig N10: 4323
	Contig N20: 2946
	Contig N30: 2166
	Contig N40: 1619
	Contig N50: 1189

	Median contig length: 388
	Average contig: 722.92
	Total assembled bases: 290845329


#####################################################
## Stats based on ONLY LONGEST ISOFORM per 'GENE':
#####################################################

	Contig N10: 3483
	Contig N20: 2214
	Contig N30: 1523
	Contig N40: 1057
	Contig N50: 733

	Median contig length: 316
	Average contig: 549.73
	Total assembled bases: 129274787
</code></pre>

---

Will likely run the resulting assembly through Trinnotate and Transdecoder to try to get a more refined assembly.

Will also run BUSCO on the refined assembly to evaluate its completeness.

Will also explore combining all of the geoduck tissue-specific transcriptome assemblies using DRAP (mentioned/suggested by Katherine in that GitHub issue).
