---
author: kubu4
comments: true
date: 2015-02-20 18:34:48+00:00
layout: post
slug: bioanalyzer-c-gigas-sheared-dna-from-20140108
title: Bioanalyzer - C.gigas Sheared DNA from 20140108
wordpress_id: 508
author:
  - kubu4
categories:
  - Lineage-specific DNA methylation patterns in developing oysters
tags:
  - bioanalyzer
  - Crassostrea gigas
  - DNA
  - DNA1000
  - electropherogram
  - heat shock
  - mantle
  - Pacific oyster
  - spreadsheet
---

To complement MBD ChiP-seq data and RNA-seq data that we have from this experiment, we want to generate, at a minimum, some BS-seq data from the same _C.gigas_ individuals used for the other aspects of this experiment.  [Claire had previously isolated DNA and sheared the DNA on 20140108](https://www.evernote.com/pub/che625/che625snotebook#st=p&n=b0e96db7-e001-411d-9870-873dad00afee). If possible, we'd like to perform MBD enrichment, but the current quantities of DNA may prevent us from this.

To quantify the DNA and evaluate the shearing profile, I ran 1μL of each of the following mantle pre-/post-heat shock samples on a DNA 1000 chip (Agilent) on the Agilent 2100 Bioanalyzer. in the Seeb Lab:

M = mantle
HS = heat shocked




    
  * 2M sheared

    
  * 4M sheared

    
  * 6M sheared

    
  * 2M HS sheared

    
  * 4M HS sheared

    
  * 6M HS sheared



Results:

Bioanalyzer Data File (XAD): [2100_expert_DNA_1000_DE72902486_2015-02-19_11-32-35(2).xad](2100%20expert_DNA%201000_DE72902486_2015-02-19_11-32-35%20(2).xad)



Electropherograms

[caption id="" align="alignnone" width="1032"]![](https://eagle.fish.washington.edu/Arabidopsis/Bioanalyzer%20Data/20150219_bioanalyzerDNA1000_claireGigasHSsheared_elecrophergrams.jpg)(http://eagle.fish.washington.edu/Arabidopsis/Bioanalyzer%20Data/20150219_bioanalyzerDNA1000_claireGigasHSsheared_elecrophergrams.jpg) 2100 Bioanalyzer electropherograms of Claire's sheared C.gigas DNA.[/caption]



Spreadsheet: [2100 expert_DNA 1000_DE72902486_2015-02-19_11-32-35_Results_001](https://docs.google.com/spreadsheets/d/1WQTZcRZ9vLjgKus3SK1Wu1JkeJN7-oPFyK-H8HiY-aU/edit?usp=sharing)

``



[Claire's notebook entry](https://www.evernote.com/pub/che625/che625snotebook#st=p&n=b0e96db7-e001-411d-9870-873dad00afee) doesn't ever specify what her target shear size was, but the Bioanalyzer analysis suggests an average size of ~500bp.

Also interesting to note is that Claire's sample concentrations (as measured on the NanoDrop1000) are significantly greater than what is calculated by the Bioanalyzer. Since the Bioanalyzer chip used (DNA1000) only goes to 1000bp, is it possible the differences in concentrations is due to incomplete shearing of the samples (e.g. a significant portion of the DNA is >1000bp in size and thus not factored in to the Bioanlyzer concentrations calculations)?

Will check sample volumes and determine total amount of remaining DNA for each sample and then assess how to proceed next (i.e. MBD or just BS-seq).

UPDATE 20150226:

Sample volumes were measured and total quantity (ng) of DNA in each sample were added to the spreadsheet above.

Based on the quantities of DNA we have for each sample, will discuss sequencing options (e.g. MBD or not, self-prepare libraries or not, etc) with Steven.


