---
author: kubu4
comments: true
date: 2018-09-19 14:05:31+00:00
layout: post
slug: transcriptome-assembly-olympia-oyster-rnaseq-data-with-trinity
title: Transcriptome Assembly - Olympia oyster RNAseq Data with Trinity
wordpress_id: 3571
author:
  - kubu4
categories:
  - Miscellaneous
tags:
  - assembly
  - mox
  - olympia oyster
  - Ostrea lurida
  - RNAseq
  - trinity
---

Used all of our current oly RNAseq data to assemble a transcriptome using Trinity.

Trinity was run our our Mox HPC node.

Reads were trimmed using the built-in version of Trimmomatic with the default settings.

SBATCH script:





  * [20180827_oly_trinity.sh](https://owl.fish.washington.edu/Athaliana/20180827_trinity_oly_RNAseq/20180827_oly_trinity.sh)



Despite the naming conventions, this job was submitted to the Mox scheduler on 201800912 and finished on 20180913.

After job completion, the entire folder was gzipped, using an interactive node (the following method of gzipping is SUPER fast, btw):


    
    <code>tar -c 20180827_trinity_oly_RNAseq | pigz > 20180827_trinity_oly_RNAseq.tar.gz</code>





* * *





#### RESULTS:



Output folder:





  * [20180827_trinity_oly_RNAseq/](https://owl.fish.washington.edu/Athaliana/20180827_trinity_oly_RNAseq/)



Trinity assembly (FastA):



  * [20180827_trinity_oly_RNAseq/Trinity.fasta](https://owl.fish.washington.edu/Athaliana/20180827_trinity_oly_RNAseq/Trinity.fasta)



Next up, I'll [follow up on this GitHub issue](httpss://github.com/RobertsLab/resources/issues/356) and get some bedgraphs generated.
