---
author: kubu4
comments: true
date: 2018-08-16 21:42:07+00:00
layout: post
slug: dna-methylation-analysis-bismark-pipeline-on-all-olympia-oyster-bsseq-datasets
title: DNA Methylation Analysis - Bismark Pipeline on All Olympia oyster BSseq Datasets
wordpress_id: 3511
author:
  - kubu4
categories:
  - Miscellaneous
tags:
  - bismark
  - BS-seq
  - methylation
  - olympia oyster
  - Ostrea lurida
---

Bismark analysis of all of our current Olympia oyster (_Ostrea lurida_) DNA methylation high-throughput sequencing data.

Analysis was run on Emu (Ubuntu 16.04LTS, Apple Xserve). The primary analysis took ~14 days to complete.

All operations are documented in a Jupyter notebook (GitHub):





  * [20180709_emu_oly_methylation_mapping.ipynb](httpss://github.com/RobertsLab/code/blob/master/notebooks/sam/20180709_emu_oly_methylation_mapping.ipynb)



Genome used:



  * [Olurida_v080.fa](https://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v080.fa) ( run was initiated prior to creation of v081; see [Genomic Resources wiki for more info ](https://github.com/RobertsLab/resources/wiki/Genomic-Resources#ostrea-lurida))





* * *



Input files ( see [Olympia oyster Genomic GitHub wiki](httpss://github.com/RobertsLab/project-olympia.oyster-genomic/wiki) for more info ):



##### WG BSseq of Fidalgo Bay offspring grown in Fidalgo Bay & Oyster Bay







  * 1_ATCACG_L001_R1_001.fastq.gz



  * 2_CGATGT_L001_R1_001.fastq.gz



  * 3_TTAGGC_L001_R1_001.fastq.gz



  * 4_TGACCA_L001_R1_001.fastq.gz



  * 5_ACAGTG_L001_R1_001.fastq.gz



  * 6_GCCAAT_L001_R1_001.fastq.gz



  * 7_CAGATC_L001_R1_001.fastq.gz



  * 8_ACTTGA_L001_R1_001.fastq.gz






##### MBDseq of two populations (Hood Canal & Oyster Bay) grown in Clam Bay







  * zr1394_10_s456.fastq.gz



  * zr1394_11_s456.fastq.gz



  * zr1394_12_s456.fastq.gz



  * zr1394_13_s456.fastq.gz



  * zr1394_14_s456.fastq.gz



  * zr1394_15_s456.fastq.gz



  * zr1394_16_s456.fastq.gz



  * zr1394_17_s456.fastq.gz



  * zr1394_18_s456.fastq.gz



  * zr1394_1_s456.fastq.gz



  * zr1394_2_s456.fastq.gz



  * zr1394_3_s456.fastq.gz



  * zr1394_4_s456.fastq.gz



  * zr1394_5_s456.fastq.gz



  * zr1394_6_s456.fastq.gz



  * zr1394_7_s456.fastq.gz



  * zr1394_8_s456.fastq.gz



  * zr1394_9_s456.fastq.gz






* * *





#### RESULTS:





With Bismark complete, these two sets of analyses can now be looked into further (and separately, as they are separate experiments) using things like MethylKit (R package) and 
the Integrative Genomics Viewer (IGV).

Output folder:





  * [owl/Athaliana/20180709_oly_methylseq](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/)



Bismark Summary Report:



  * [20180709_oly_methylseq/bismark_summary_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/bismark_summary_report.html)



Individual Sample Reports:



  * [1_ATCACG_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/1_ATCACG_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [2_CGATGT_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/2_CGATGT_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [3_TTAGGC_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/3_TTAGGC_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [4_TGACCA_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/4_TGACCA_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [5_ACAGTG_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/5_ACAGTG_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [6_GCCAAT_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/6_GCCAAT_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [7_CAGATC_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/7_CAGATC_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [8_ACTTGA_L001_R1_001_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/8_ACTTGA_L001_R1_001_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_10_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_10_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_11_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_11_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_12_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_12_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_13_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_13_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_14_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_14_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_15_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_15_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_16_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_16_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_17_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_17_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_18_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_18_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_1_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_1_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_2_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_2_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_3_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_3_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_4_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_4_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_5_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_5_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_6_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_6_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_7_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_7_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_8_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_8_s456_trimmed_bismark_bt2_SE_report.html)



  * [zr1394_9_s456_trimmed_bismark_bt2_SE_report.html](https://owl.fish.washington.edu/Athaliana/20180709_oly_methylseq/zr1394_9_s456_trimmed_bismark_bt2_SE_report.html)



