---
author: kubu4
comments: true
date: 2018-09-25 23:46:26+00:00
layout: post
slug: transcriptome-alignment-olympia-oyster-rnaseq-reads-aligned-to-genome-with-hisat2
title: Transcriptome Alignment â€“ Olympia oyster RNAseq reads aligned to genome with
  HISAT2
wordpress_id: 3586
author:
  - kubu4
categories:
  - Miscellaneous
tags:
  - alignment
  - hisat2
  - mox
  - olympia oyster
  - Ostrea lurida
---

[Yesterday's attempt at producing a bedgraph](http://onsnetwork.org/kubu4/2018/09/24/bedgraph-olympia-oyster-transcriptome-fail/) was a failure and a prodcuct of a major brain fart. The worst part is that I was questioning what I was doing the entire time, but still went through with the process! Yeesh!

The problem was that I tried to take our Trinity-assembled transcriptome and somehow align that to our genome. This can't work because each of those assemblies don't know the coordinates used by the other. So, as was the case, you end up with a bedgraph that shows zero coverage for all genome contigs.

Anyway, here's the correct procedure!

Used HISAT2 on our HPC Mox node to align our RNAseq reads to our Olurida_v081 genome assembly:





  * [Olurida_v081.fa](http://owl.fish.washington.edu/halfshell/genomic-databank/Olurida_v081.fa)



SBATCH script files:

PERFORM GENOME INDEXING & ALIGNMENT
- [20180925_oly_RNAseq_genome_hisat2.sh](http://owl.fish.washington.edu/Athaliana/20180925_oly_RNAseq_genome_hisat2/20180925_oly_RNAseq_genome_hisat2.sh)

SORT & INDEX ALIGNMENT OUTPUT
- [20180925_oly_RNAseq_genome_sort_bam.sh](20180925_oly_RNAseq_genome_hisat2.sh)



* * *





#### RESULTS



Output folder:





  * [20180925_oly_RNAseq_genome_hisat2/](http://owl.fish.washington.edu/Athaliana/20180925_oly_RNAseq_genome_hisat2/)



Sorted BAM file (58GB):



  * [20180925_Olurida_v081.sorted.bam](http://owl.fish.washington.edu/Athaliana/20180925_oly_RNAseq_genome_hisat2/20180925_Olurida_v081.sorted.bam)



Will get the sorted BAM file converted to a bedgraph showing genome coverage for use in IGV.
