---
author: kubu4
comments: true
date: 2018-08-30 22:23:12+00:00
layout: post
link: http://onsnetwork.org/kubu4/2018/08/30/fastqcmultiqctrimgaloremultiqcfastqcmultiqc-o-lurida-wgbsseq-for-methylation-analysis/
slug: fastqcmultiqctrimgaloremultiqcfastqcmultiqc-o-lurida-wgbsseq-for-methylation-analysis
title: FastQC/MultiQC/TrimGalore/MultiQC/FastQC/MultiQC - O.lurida WGBSseq for Methylation
  Analysis
wordpress_id: 3554
author:
  - kubu4
categories:
  - Miscellaneous
tags:
  - BSseq
  - FASTQC
  - mox
  - multiqc
  - olympia oyster
  - Ostrea lurida
  - TrimGalore!
  - trimming
---

I previously ran this data [through the Bismark pipeline](http://onsnetwork.org/kubu4/2018/08/16/dna-methylation-analysis-bismark-pipeline-on-all-olympia-oyster-bsseq-datasets/) and [followed up with MethylKit analysis](http://onsnetwork.org/kubu4/2018/09/04/dna-methylation-analysis-olympia-oyster-bsseq-methylkit-analysis/). MethylKit analysis revealed an extremely low number of differentially methylated loci (DML), which seemed odd.

Steven and I met to discuss and compare our different variations on the analysis and decided to try out different tweaks to evaluate how they affect analysis.

I did the following tasks:





  1. Looked at original sequence data quality with FastQC.



  2. Summarized FastQC analysis with MultiQC.



  3. Trimmed data using TrimGalore!, trimming 10bp from 5' end of reads (8bp is recommended by Bismark docs).



  4. Summarized trimming stats with MultiQC.



  5. Looked at trimmed sequence quality with FastQC.



  6. Summarized FastQC analysis with MultiQC.






This was run on the Univ. of Washington High Performance Computing (HPC) cluster, Mox.

Mox SBATCH submission script has all details on how the analyses were conducted:





  * [20180830_oly_WGBSseq_trimming.sh](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_oly_WGBSseq_trimming.sh)





* * *





##### RESULTS



Output folder:





  * [20180830_oly_WGBSseq_trimming/](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/)



Raw sequence FastQC output folder:



  * [20180830_oly_WGBSseq_trimming/20180830_fastqc/](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_fastqc/)



Raw sequence MultiQC report (HTML):



  * [20180830_oly_WGBSseq_trimming/20180830_fastqc/multiqc_report.html](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_fastqc/multiqc_report.html)



TrimGalore! output folder (trimmed FastQ files are here):



  * [20180830_oly_WGBSseq_trimming/20180830_trimgalore/](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_trimgalore/)



Trimming MultiQC report (HTML):



  * [20180830_oly_WGBSseq_trimming/20180830_trimgalore/multiqc_report.html](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_trimgalore/multiqc_report.html)



Trimmed FastQC output folder:



  * [20180830_oly_WGBSseq_trimming/20180830_trimmed_fastqc/](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_trimmed_fastqc/)



Trimmed MultiQC report (HTML):



  * [20180830_oly_WGBSseq_trimming/20180830_trimmed_fastqc/multiqc_report.html](http://owl.fish.washington.edu/Athaliana/20180830_oly_WGBSseq_trimming/20180830_trimmed_fastqc/multiqc_report.html)


